lib/status-bar-manager.coffee

```diff
@@ -14,9 +19,10 @@ class StatusBarManager
 
   initialize: (@statusBar) ->
 
-  update: (currentMode) ->
+  update: (currentMode, currentSubmode) ->
+    currentFullMode = currentMode + "$" + currentSubmode if currentSubmode?
```

Comment:
Oh, I see what you're saying. I think the most straightforward thing to do is to use the code snippet I wrote above; we just need to enumerate *all* possible combinations of `mode` and `submode` in `ContentsByMode`, rather than abbreviating `visual.characterwise` as `visual`.

I don't think we can use a `break` statement like you have on line 30, because it could incorrectly prevent classes from being removed from the `@element` div.